src/tomahawk/TomahawkWindow.cpp

```diff
@@ -757,11 +757,23 @@ TomahawkWindow::changeEvent( QEvent* e )
 }
 
 
+// CHANGED
 void
 TomahawkWindow::closeEvent( QCloseEvent* e )
 {
+
+    // CHANGED change according to definitions
+    TomahawkSettings* s = TomahawkSettings::instance();
+    //QMainWindow::setQuitOnLastWindowClosed(s->exitOnClose());
+
+	if(s->exitOnClose()){
+        qApp->quit();
+        return;
+    }
+
 #ifndef Q_OS_MAC
-    if ( e->spontaneous() && QSystemTrayIcon::isSystemTrayAvailable() )
+    // CHANGED
+    if (!s->exitOnClose() && e->spontaneous() && QSystemTrayIcon::isSystemTrayAvailable())
```

Comment:
a) you removed whitespaces that we like.
b) is this even neccessary? If the diff view in Github isn't fooling me, you're `return`ing anyway if `exitOnClose()` is true, so this code is never able to run. To make it more obvious you could make it `else if()` in the `Q_OS_MAC` section.